### ISSUE TITLE
[TF2 - Feature Request] Allow party-leaders to enable/disable autobalance over the whole party in casual matches.

### ISSUE THREAD

**Issue by Sphirye** [ValveSoftware/Source-1-Games#5337]
I've been pondering the pros and cons of an implementation that allows party leaders to have a checkbox on ther party settings that allow auto-balance over the other party members.

### Why a checkbox and not just disabling auto-balance on parties in general?

Most likely because dual parties, its common to find two friends players who take advantage of auto-balance to play againsts and even challenge each other to duels.

# Pros
### 1. A better gameplay experience.
I "understand" the existence of the current auto-balance system, but not only being balanced, but getting separed from your friends is just a too bitter game feeling.

# Cons
Most of these are kind of situational cons and open to debate, since these are more related to the current TF2 status than the function itself:

### 1. Cheaters
There are casual cheaters that play with their friends using parties to prevent getting kicked themselves, however, there are times where one of the cheaters (or the cheater friends) gets balanced, allowing the other team to kick it.
(which would count if only tf2 or steam would allow you to see which the players are together in a party).

_This is very circumstantial, since there are already resources that can bypass auto-balance by forcing the player to restart the game when they detect an upcoming one._

### 2. Implementation Integrity
This isn't even an actual con, its just a warning.

An enhancement like this is fundamentally designed so that its use is restricted exclusively to players who are already in a party, so it will most likely be programmed as a another party setting.

If the TF2 current party system has some strange vulnerability, its wukk most likely to get abused, where in the worst it would even allow people to bypass auto-balance even not being in a party.

**Comment by gaelcoral** [comment]
I agree with this, especially with the contracts

**Comment by Wosheereal** [comment]
No one tell em that this was planned after jungle inferno but never got implemented.

**Comment by okamiterasu** [comment]
Yeah, sure. Just remove the only remaining safeguard against team stacking.

**Comment by Sphirye** [comment]
> Yeah, sure. Just remove the only remaining safeguard against team stacking.

I'm not asking for removing anything, read the post again.

Even if what you said was true, parties have a maximum of 6 players. For this supposed stacking to occur a team would need to have exactly two parties of six players, that is just unrealistic.

**Comment by okamiterasu** [comment]
> Even if what you said was true, parties have a maximum of 6 players. For this supposed stacking to occur a team would need to have exactly two parties of six players, that is just unrealistic.

You don't need to stack the entire team to cause problems. Six people is more than enough to ruin a match

>I'm not asking for removing anything, read the post again.

Sure, we'll just politely ask people to not abuse the option. After all, things like the autobalance volunteer system were renowned for their effectiveness.

**Comment by Sphirye** [comment]
>You don't need to stack the entire team to cause problems. Six people is more than enough to ruin a match

Then thats a team skill matching issue. This feature request focuses on not separing players from their party.

If you are worried about team-stacking then go to #4253 or open your own issue asking to enhance the current matchmaking system, because i honestly dont think randomly moving a player out of their pro six-party team will magically fix a match.
